pkg/k8s/pipelineRun.go

```diff
@@ -126,20 +126,14 @@ func (r *pipelineRun) UpdateState(state api.State) (*api.StateItem, error) {
 	r.ensureCopy()
 	log.Printf("New State: %s", state)
 	now := metav1.Now()
-	oldstate, err := r.FinishState()
-	if err != nil {
-		return nil, err
-	}
+	oldstate := r.finishCurrentState()
 	newState := api.StateItem{State: state, StartedAt: now}
 	r.apiObj.Status.StateDetails = newState
 	r.apiObj.Status.State = state
 	return oldstate, r.updateStatus()
 }
 
-// FinishState set end time stamp of the current (defined) state and add it to the history
-// If no current state is defined a new state (A) with creation time of the PipelineRun as start time is created.
-// Returns the state details
-func (r *pipelineRun) FinishState() (*api.StateItem, error) {
+func (r *pipelineRun) finishCurrentState() *api.StateItem {
```

Comment:
The func names do not explain well what's the difference.
In another project we used names like `finishStateUncommitted`. `setFinishState` or `prepareFinishState` would also be an option.

Reply:
While reading the code and usage `finishCurrentState` makes more sense... name could be OK.